src/btree.h

```diff
@@ -147,6 +147,11 @@ typedef struct btree_iterator {
    debug_code(char debug_prev_end_key[MAX_KEY_SIZE]);
 } btree_iterator;
 
+typedef enum btree_pack_err {
```

Comment:
If you have `btree_pack_loop` return the error instead of sticking it inside the `req`, then you can move this enum into `btree.c`.  It doesn't need to be part of the interface to the btree.